src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -1444,6 +1556,41 @@ nns_edge_start (nns_edge_h edge_h)
   return ret;
 }
 
+/**
+ * @brief Stop the nnstreamer edge.
+ */
+int
+nns_edge_stop (nns_edge_h edge_h)
+{
+  nns_edge_handle_s *eh;
+  int ret = NNS_EDGE_ERROR_NONE;
+
+  eh = (nns_edge_handle_s *) edge_h;
+  if (!eh) {
+    nns_edge_loge ("Invalid param, given edge handle is null.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!nns_edge_handle_is_valid (eh)) {
+    nns_edge_loge ("Invalid param, given edge handle is invalid.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  nns_edge_lock (eh);
+
+  if (NNS_EDGE_CONNECT_TYPE_CUSTOM == eh->connect_type) {
+    NnsEdgeCustomDef *custom_h = (NnsEdgeCustomDef *) eh->custom_h;
+    ret = custom_h->nns_edge_custom_stop (eh->broker_h);
+    if (NNS_EDGE_ERROR_NONE != ret) {
+      nns_edge_loge ("Failed to stop MCF");
+    }
+  }
```

Comment:
The nns stop behavior is well programmed according to NNS_EDGE_CONNECT_TYPE_CUSTOM. :+1: 
How about changing this part to this?
```
ret = handle_connection_type(edge_handle);
```
```
static int handle_connection_type(nns_edge_handle_s* edge_handle) {
  int ret = NNS_EDGE_ERROR_NONE;

  if (edge_handle->connect_type == NNS_EDGE_CONNECT_TYPE_CUSTOM) {
    NnsEdgeCustomDef* custom_handle = (NnsEdgeCustomDef*) edge_handle->custom_h;
    ret = custom_handle->nns_edge_custom_stop(edge_handle->broker_h);
    if (ret != NNS_EDGE_ERROR_NONE) {
      nns_edge_loge("Failed to stop MCF.");
    }
  }

  return ret;
}
```

Reply:
Thank you. Thanks for the great code. 💯